### ISSUE TITLE
Beatmap show as 0 star, also freeze on result screen if happend.

### ISSUE THREAD

**Issue by IkaWaAyuMu** [ppy/osu#11094]
**Describe the crash:** Beatmap sometimes show as 0 star (other difficulty shown normally) and if open result screen in any way (TESTED : end playing, open other player result) the game freeze and never self-close.
Update : Exact reproduction by @Joehuu https://github.com/ppy/osu/issues/11094#issuecomment-738970361

**Screenshots or videos showing encountered issue:** https://streamable.com/pmv64a

**osu!lazer version:** 2020.1204.0

**Logs:**
[database.log](https://github.com/ppy/osu/files/5643045/database.log)
[network.log](https://github.com/ppy/osu/files/5643046/network.log)
[performance.log](https://github.com/ppy/osu/files/5643047/performance.log)
[runtime.log](https://github.com/ppy/osu/files/5643048/runtime.log)
[updater.log](https://github.com/ppy/osu/files/5643049/updater.log)

(Yes, the log doesn't provide any error.)

**Computer Specifications:** 
CPU : Intel i5-6600K
GPU : Nvidia GTX 960
RAM : 8 GB

Beatmap directory is located on HDD. Game directory is located on SSD.

**Comment by bdach** [comment]
Please link the map.

**Comment by IkaWaAyuMu** [comment]
> Please link the map.

Happened to all maps after the happened and can happened to any maps.
So I think I don't have to provided.

Btw the map/diff shown is https://osu.ppy.sh/beatmapsets/1292150#osu/2681790

**Comment by bdach** [comment]
It's most likely triggered by one particular map, at which point star calculation will fail for all subsequent maps.

**Comment by IkaWaAyuMu** [comment]
> It's most likely triggered by one particular map, at which point star calculation will fail for all subsequent maps. Hence the question.

Oh. It happened before I select this map.
So I think this map isn't the case, but I record after I randomized to this map.

**Comment by bdach** [comment]
Let me rephrase: I think the most likely failure scenario is that on every launch, star rating will work until you select some particular map, after which it will stop working for all maps that will be selected from that point on.

Please confirm whether that is what you're seeing and either try to find the map that breaks star ratings, or provide `client.db` so that I can look through your map collection and try to find it myself. Either way there is not enough info to begin investigating this right now.

**Comment by IkaWaAyuMu** [comment]
> Let me rephrase: I think the most likely failure scenario is that on every launch, star rating will work until you select some particular map, after which it will stop working for all maps that will be selected from that point on.
> 
> Please confirm whether that is what you're seeing and either try to find the map that breaks star ratings, or provide `client.db` so that I can look through your map collection and try to find it myself. Either way there is not enough info to begin investigating this right now.

[client.db](https://doc-0k-6o-docs.googleusercontent.com/docs/securesc/0vgn814par8vhugjj4spgub7u2lqkfsg/r82il7r1kg2a7o62qunlnbanf20ombg0/1607092350000/14331776476481591563/14331776476481591563/1Q02J66O4xfHmIqbsjfZo0YIBGjXxTXA8?e=download&authuser=0&nonce=e2prhsu0mjati&user=14331776476481591563&hash=pa3bfd3d7i3h7cr0ssr3hp5oa1rbv1sm)

I can't find "that" map right now, so I give you `client.db` first

**Comment by Joehuu** [comment]
Can reproduce with any decent amount of maps on each mode and show converted disabled. You just need to spam between modes and go though maps that are not calculated yet. Also score calculation seems to break.

![LWIVMIn2GU](https://user-images.githubusercontent.com/35318437/101205575-ac27a780-3622-11eb-8ebf-68b15cf427eb.gif)

**Comment by IkaWaAyuMu** [comment]
> Can reproduce with any decent amount of maps on each mode and show converted disabled. You just need to spam between modes and go though maps that are not calculated yet. Also score calculation seems to break.
> 
> ![LWIVMIn2GU](https://user-images.githubusercontent.com/35318437/101205575-ac27a780-3622-11eb-8ebf-68b15cf427eb.gif)

Just note :  **gameplay** score calculation seems fine

[Normal standardized](https://streamable.com/8w8c9e)
[Bugged standardized](https://streamable.com/rlv6kn)

[Normal classic](https://streamable.com/teslkq)
[Bugged classed](https://streamable.com/9cqyyr)

[[Map and diff I use]](https://osu.ppy.sh/beatmapsets/1124845#osu/2542146)

**Comment by Joehuu** [comment]
After bisecting, this seems to start happening after https://github.com/ppy/osu/pull/10712.

**Comment by peppy** [comment]
@Joehuu do you see any error in your logs when this happens? I can reproduce the missing star difficulty but have not got the crash yet.

**Comment by Joehuu** [comment]
Nope, no error in logs. There are two bugs at play here. The first one you reproduced, and the second one is going to results screen via the leaderboard or end of gameplay (which causes rhe weird crash).

**Comment by peppy** [comment]
I managed to get crashes with no error, but it looked to be to be a gpu driver issue. I'm wondering if this is the same thing occuring for you guys as well in this case...

The only other possibility I can see happening is that an unobserved exception is being fired (there is at least one location in the `BeatmapDifficultyCache` component where I found this can occur). I might look at covering those as a first step.
